docs/spec/spec.md

```diff
@@ -0,0 +1,694 @@
+This library is programmed in the [Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the cloud
+that makes it easier to use, combine, and create network services.
+
+# Contents
+
+1. [Overview](#1-overview)  
+2. [Configurations](#2-configurations)  
+   2.1 [Producer Configurations](#21-producer-configurations)   
+   2.2 [Consumer/Listener Configurations](#22-consumerlistener-configurations)    
+   2.3 [Security Configurations](#23-security-configurations)      
+   2.4 [Common Configurations](#24-common-configurations)   
+   *   2.4.1 [TopicPartition](#241-topicpartition)     
+   *   2.4.2 [PartitionOffset](#242-partitionoffset)   
+3. [Producer](#3-producer)  
+   3.1 [Connection](#31-connection)     
+   *    3.1.1 [Insecure Connection](#311-insecure-connection)   
+   *    3.1.2 [Secure Connection](#312-secure-connection)   
+   3.2 [Producer Usage](#32-producer-usage)     
+4. [Consumer](#4-consumer)  
+   4.1 [Connection](#41-connection)  
+   *    4.1.1 [Insecure Connection](#411-insecure-connection)  
+   *    4.1.2 [Secure Connection](#412-secure-connection)   
+   4.2 [Consumer Usage](#42-consumer-usage)    
+   *    4.2.1 [Consume Messages](#421-consume-messages)  
+   *    4.2.2 [Handle Offsets](#422-handle-offsets)    
+   *    4.2.3 [Handle Partitions](#423-handle-partitions)    
+   *    4.2.4 [Seeking](#424-seeking)    
+   *    4.2.5 [Handle subscriptions](#425-handle-subscriptions)   
+5. [Listener](#5-listener)  
+   5.1 [Connection](#51-connection)     
+   *    5.1.1 [Insecure Connection](#511-insecure-connection)  
+   *    5.1.2 [Secure Connection](#512-secure-connection)   
+   5.2 [Listener Usage](#52-listener-usage)    
+   *    5.2.1 [Caller](#521-caller)    
+6. [Samples](#6-samples)      
+   6.1 [Produce Messages](#61-produce-messages)    
+   6.2 [Consume Messages](#62-consume-messages)    
+   *    6.2.1 [Using Consumer](#621-using-consumer)   
+   *    6.2.2 [Using Listener](#622-using-listener)   
+
+## 1. Overview
+
+Apache Kafka is an open-source distributed event streaming platform for high-performance data pipelines, 
+streaming analytics, data integration, and mission-critical applications. This specification elaborates on 
+the usage of Kafka clients, producer and consumer. These clients allow writing distributed applications and 
+microservices that read, write, and process streams of events in parallel, at scale, and in a fault-tolerant 
+manner even in the case of network problems or machine failures.
+
+Ballerina Kafka contains three core apis:
+* Producer - Used to publish messages to the Kafka server.
+* Consumer & Listener - Used to get the messages from the Kafka server.
+
+## 2. Configurations
+### 2.1 Producer Configurations
+When initializing the producer, following configurations can be provided.
+```ballerina
+public type ProducerConfiguration record {|
```

Comment:
It would be great if we can have the doc comments above each configuration. I think we already have it in the code.

Reply:
Updated